game-core/src/main/java/games/strategy/engine/display/IDisplay.java

```diff
@@ -54,6 +55,8 @@ void reportMessageToPlayers(
    * @param attacker - PlayerId of attacker
    * @param defender - PlayerId of defender
    */
+  @RemoveOnNextMajorRelease(
+      "Remove isAmphibious, amphibiousLandAttackers, dependentUnits, and battleTitle")
   @RemoteActionCode(12)
   void showBattle(
```

Comment:
This is only for showing the battle display.  It actually makes a copy of all of these parameters (see `BattlePanel#showBatle`).

It looks like the actual battle takes place on the client that is attacking.  The other clients have the battle display and this is initially populated with the data from the attacking client and then updated through calls from the attacking client.

This method looks like it could use the `BattleState` instead of the individual items.